[Component] Navdots

fallaciousreasoning opened the pull request:
Fixes https://github.com/brave/leo/issues/65

Add navigation dots
[Screencast from 2022-11-28 12-34-57.webm](https://user-images.githubusercontent.com/7678024/204165735-52252112-89e6-4a8d-8176-6ed5e6d6f785.webm)

Storybook:
https://leo-storybook-git-wc-nav-dots-fallaciousreasoning.vercel.app/?path=/story/nav-dots--default

aguscruiz commented:
@fallaciousreasoning should I wait for the latest updates to review? Let me know

aguscruiz commented:
Minor thing that's not gonna be in the design, the cursor should be the pointer for these buttons (and in general, I think a good rule is making all the clickeable elements have the pointer cursor)

fallaciousreasoning commented:
Nah, I think any feedback you have will be useful now (the cursor is a good catch, I completely forgot :D)

aguscruiz commented:
Ok, I'll add it here then: 

- The inactive dot should be 8px wide/tall. And has a 1px margin, so it's 10px wide/tall counting it.
- Active dot is 10px tall, a bit bigger than the inactive, and 18px wide.
- Margin between dots (without counting the active dot) is 8px.
- Adjusted the hover states in the design based on your comments.
- Have the cursor be pointer
- I think the animation works perfect here 

<img width="325" alt="image" src="https://user-images.githubusercontent.com/15734630/204362823-28c5fffc-1984-46ee-ae98-ca7d25782fcd.png">
- I don't know if this nav margin is a default or just for storybook, but I'd suggest not adding it by default because it may change depending on the design.

fallaciousreasoning commented:
>- The inactive dot should be 8px wide/tall. And has a 1px margin, so it's 10px wide/tall counting it.

Whoops - I made it double size to make it easier to test and forgot to change it back!

> - Active dot is 10px tall, a bit bigger than the inactive, and 18px wide.

I completely missed the fact it was taller than the dots! Fixed.

>- Adjusted the hover states in the design based on your comments.

Cheers, thank you!

>- Have the cursor be pointer

Nice catch!

>- I think the animation works perfect here 

:heart: 

>- I don't know if this nav margin is a default or just for storybook, but I'd suggest not adding it by default because it may change depending on the design.

Oooh, didn't catch that. I swapped to using an `<ol>` tag for accessibility and I think the margin came with it. Removed!

petemill commented on web-components/navdots/navdots.stories.svelte:
```diff
@@ -0,0 +1,28 @@
+/>
+
+<Template let:args>
+  <div style={`--color-interaction-button-primary-background: ${args.activeColor || 'unset'}; --color-primary-20: ${args.defaultColor || 'unset'}`}>
```
What do you think about making a component-specific variable that can be overriden, e.g. `--color-navdots`. I worry about the parents knowing the specific variable that gets used.

petemill replied:
Separately, and this is also a question for @aguscruiz - do we need to allow overriding _active_ dot color as well as dot color, or is there some kind of consistent calculation from dot-color to active-dot-color, like opacity?

fallaciousreasoning replied:
Yeah, it'd be really neat if everything derived from one color :smile:

fallaciousreasoning commented:
- Move CSS variables to root & make them global
- Maybe use a span for the active indicator

petemill commented on web-components/navdots/navdots.svelte:
```diff
@@ -0,0 +1,127 @@
+        background-color: var(--color-primary-30);
+      }
+
+      &:focus-visible:not(.active) {
```
let's wrap this with 
```
@supports selector(:has(a)) {
}
```

petemill commented on web-components/navdots/navdots.svelte:
```diff
@@ -0,0 +1,127 @@
+        display: flex;
+      }
+
+      &:has(.dot.active:focus-visible) .active-dot {
```
let's wrap this with 
```
@supports selector(:has(a)) {
}
```
